src/software/backend/robot_status.cpp

```diff
@@ -0,0 +1,82 @@
+#include "software/backend/robot_status.h"
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "software/logger/logger.h"
+
+TbotsRobotMsg convertRobotStatusToTbotsRobotMsg(RobotStatus* robot_status)
+{
+    std::map<std::string, ErrorCode> errorStringToEnumMap = {
+        {"charge timeout", CHARGE_TIMEOUT},
+        {"wheel 0 motor hot", WHEEL_0_MOTOR_HOT},
+        {"wheel 1 motor hot", WHEEL_1_MOTOR_HOT},
+        {"wheel 2 motor hot", WHEEL_2_MOTOR_HOT},
+        {"wheel 3 motor hot", WHEEL_3_MOTOR_HOT},
+        {"dribbler motor hot", DRIBBLER_MOTOR_HOT},
+        {"wheel 0 encoder not commutating", WHEEL_0_ENCODER_NOT_COMMUTATING},
+        {"wheel 1 encoder not commutating", WHEEL_1_ENCODER_NOT_COMMUTATING},
+        {"wheel 2 encoder not commutating", WHEEL_2_ENCODER_NOT_COMMUTATING},
+        {"wheel 3 encoder not commutating", WHEEL_3_ENCODER_NOT_COMMUTATING},
+        {"wheel 0 Hall sensor stuck low", WHEEL_0_HALL_SENSOR_STUCK_LOW},
+        {"wheel 1 Hall sensor stuck low", WHEEL_1_HALL_SENSOR_STUCK_LOW},
+        {"wheel 2 Hall sensor stuck low", WHEEL_2_HALL_SENSOR_STUCK_LOW},
+        {"wheel 3 Hall sensor stuck low", WHEEL_3_HALL_SENSOR_STUCK_LOW},
+        {"dribbler Hall sensor stuck low", DRIBBLER_HALL_SENSOR_STUCK_LOW},
+        {"wheel 0 Hall sensor stuck high", WHEEL_0_HALL_SENSOR_STUCK_HIGH},
+        {"wheel 1 Hall sensor stuck high", WHEEL_1_HALL_SENSOR_STUCK_HIGH},
+        {"wheel 2 Hall sensor stuck high", WHEEL_2_HALL_SENSOR_STUCK_HIGH},
+        {"wheel 3 Hall sensor stuck high", WHEEL_3_HALL_SENSOR_STUCK_HIGH},
+        {"dribbler Hall sensor stuck high", DRIBBLER_HALL_SENSOR_STUCK_HIGH},
+    };
+
+    // Insufficient ChipperKickerStatus, DriveUnits, and NetworkStatus info
+    TbotsRobotMsg robot_msg;
+
+    robot_msg.set_robot_id(robot_status->robot);
+
+    for (const auto& msg : robot_status->robot_messages)
+    {
+        // If it is a valid error message, add to error_code
+        if (errorStringToEnumMap.find(msg) != errorStringToEnumMap.end())
+        {
+            robot_msg.add_error_code(errorStringToEnumMap[msg]);
+        }
+        else
+        {
+            LOG(WARNING) << "'" << msg << "' is not a verified ErrorCode." << std::endl;
+        }
+    }
```

Comment:
I think we should be able to get rid of this check if we switch to the Enum suggestion I had above?

Reply:
The purpose of the LOG is because `robot_messages` may also have messages that aren't ErrorCode messages. So it might still be a good idea to have this in there.